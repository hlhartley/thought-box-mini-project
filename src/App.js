import React, { Component } from 'react';
import './CSS/main.scss';
import CreateThought from './createThought/CreateThought';
import { ThoughtList } from './thoughtList/ThoughtList';
import Header from './Header';

class App extends Component {
  constructor() {
    super();
    this.state = {
      thoughts: []
    };
  }

  createThought = (thought) => {
    const { thoughts } = this.state
    const newThought = {...thought, id: thoughts.length}
    this.setState({ thoughts: [...thoughts, newThought] });
  }

  render() {
    return (
      <div className="app">
        <Header />
        <CreateThought createThought={this.createThought} />
        <div>
          <ThoughtList thoughtList={this.state.thoughts} />
        </div>
      </div>
    );
  }
}

export default App;
