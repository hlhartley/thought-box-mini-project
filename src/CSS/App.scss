.app {
  display: grid;
  line-height: 35px;
}

.create-thought-container,
.thought-list-container {
  background-color: #ededed;
  padding-top: 30px;
  display: flex;
  flex-flow: column;
  align-items: center;
}

.create-thought-container {
  border-bottom: 2px solid grey;
}
