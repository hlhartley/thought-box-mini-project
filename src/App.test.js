import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';
  
let wrapper;
  
describe('App', () => {

beforeEach(() => {
    wrapper = shallow(
    <App />
    );
})

    it('matches the snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('renders a thoughtList with the correct props', () => {
        const { thoughtList, searchInput, removeThought } = wrapper.find(ThoughtList).props();
        expect(thoughtList).toEqual([]);
        expect(searchInput).toEqual('');
        // expect(removeThought).toBeInstanceOf(Function);
    });

    it('should have a proper default state', () => {
        expect(wrapper.state()).toEqual({
            searchInput: "", thoughts: [],
        })
    })

    it('createThought adds a thought to state', () => {
        wrapper.instance().createThought()
        expect(wrapper.state()).toEqual({
            searchInput: "", thoughts: [{"id": 0}],
        })
    });
})
// Feel free to add additional tests. These are meant just to get you started.
