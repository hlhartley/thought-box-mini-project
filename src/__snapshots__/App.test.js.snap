// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          searchThoughts={[Function]}
        />,
        <CreateThought
          createThought={[Function]}
        />,
        <div
          className="thought-list-container"
        >
          <ThoughtList
            removeThought={[Function]}
            searchInput=""
            thoughtList={Array []}
          />
        </div>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "searchThoughts": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "createThought": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ThoughtList
            removeThought={[Function]}
            searchInput=""
            thoughtList={Array []}
          />,
          "className": "thought-list-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "removeThought": [Function],
            "searchInput": "",
            "thoughtList": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            searchThoughts={[Function]}
          />,
          <CreateThought
            createThought={[Function]}
          />,
          <div
            className="thought-list-container"
          >
            <ThoughtList
              removeThought={[Function]}
              searchInput=""
              thoughtList={Array []}
            />
          </div>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "searchThoughts": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "createThought": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ThoughtList
              removeThought={[Function]}
              searchInput=""
              thoughtList={Array []}
            />,
            "className": "thought-list-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "removeThought": [Function],
              "searchInput": "",
              "thoughtList": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
